src/content/docs/linter/index.mdx

```diff
@@ -4,180 +4,205 @@ description: How to use the Biome linter.
+import PackageManagerBiomeCommand from "@/components/PackageManagerBiomeCommand.astro";
 
-Biome's linter statically analyzes your code to catch common errors and to help writing idiomatic code.
+Biome's linter statically analyzes your code to find and fix common errors and to help you write better code.
+It [supports multiple languages](/internals/language-support) and provides a total of [**<NumberOfRules/> rules**](/linter/rules/).
 
-<NumberOfRules />
 
-## Use the linter via CLI
+## CLI
 
-You can start by running the CLI to check for possible errors using the following command:
+The following command runs the linter on all files in the `src` directory:
 
-```shell
-biome lint ./src
-```
-
-For more information about all the available options, check the [CLI page](/reference/cli#biome-lint)
+<PackageManagerBiomeCommand command="lint ./src" />
 
-## Rule pillars
-
-We believe that rules should be informative and explain to the user why a rule is triggered and tell the user what they should to do fix the error. A rule should follow these **pillars**:
-1. Explain to the user the error. Generally, this is the message of the diagnostic.
-2. Explain to the user **why** the error is triggered. Generally, this is implemented with an additional node.
-3. Tell the user what they should do. Generally, this is implemented using a code action. If a code action is not applicable a note should tell the user what they should do to fix the error.
+Note that if you pass a glob as a parameter, it will be expanded by your shell.
+The result of the expansion depends on your shell and its configuration.
+For example, some shells don't support the recursive glob `**` or the alternation `{}` in the following command:
 
-If you think a rule doesn't follow these pillars, please [file an issue](https://github.com/biomejs/biome/issues/new?assignees=&labels=S-To+triage&projects=&template=01_bug.yml&title=%F0%9F%90%9B+%3CTITLE%3E).
+```shell
+biome lint ./src/**/*.test.{js,ts}
+```
 
-## Code fixes
+For more information about all the available options, check the [CLI reference](/reference/cli#biome-lint).
 
-Lint rules may provide automatic code fixes. Biome distinguishes between two types of fixes.
 
-### Safe fixes
+## Rules
 
-Safe fixes are guaranteed to not change the semantic of your code. They can be applied without explicit review.
+The linter is organized into rules.
+A rule emit a diagnostic when it encounters a code that doesn't meet its requirements.
+For example, the [noDebugger](http://localhost:4321/linter/rules/no-debugger) rule reports the use of the `debugger` instruction in JavaScript code.
 
-To apply _safe fixes_, use `--apply`:
+A rule emits diagnostics with a `warn` or `error` severity.
+Diagnostics with an `error` severity cause the command to exit with a non-zero code,
+While diagnostics with a `warn` severity don't cause the command to fail.
+You can turn warnings into errors `--error-on-warnings` CLI option:
```

Comment:
I would change the explanation and focus more on the rules rather than discussing diagnostics. In the future, a full, in-depth explanation of our diagnostics should be on a separate page.

My suggestion:

```suggestion
By default, **recommended** rules emit a diagnostic with `error` severity and will cause the command to fail. Via configuration, you can demote recommended rules to emit warnings or promote non-recommended rules to emit errors.

You can turn warnings into errors `--error-on-warnings` CLI option:
```